provider/datagen/tests/collator-extended-data-test.rs

```diff
@@ -0,0 +1,125 @@
+    {
+        assert_eq!(collator.compare(left_str, right_str), *expectation);
+    }
+}
+
+#[test]
+fn test_sv() {
+    // This is the same as test_fi. The purpose of this copy is to test that
+    // Swedish defaults to "reformed", which behaves like Finnish "standard",
+    // and not to "standard", which behaves like Finnish "traditional".
+
+    // Adapted from ficoll.cpp in ICU4C
+    // Testing that w and v behave as in the root collation is for checking
+    // that the sorting collation doesn't exhibit the behavior of the search
+    // collation, which (somewhat questionably) treats w and v as primary-equal.
+    let left = [
+        "wat",
+        "vat",
+        "aübeck",
+        "Låvi",
+        // ICU4C has a duplicate of the first case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        "ä",
+        "a\u{0308}",
+    ];
+    let right = [
+        "vat", "way", "axbeck", "Läwe",
+        // ICU4C has a duplicate of the first case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        "o", "ä",
+    ];
+    let expectations = [
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Equal,
+    ];
+    let locale: Locale = langid!("sv").into();
+
+    let any_dyn_provider = get_provider();
+    let any_provider = any_dyn_provider.as_any_provider();
+    let provider_no_fallback = any_provider.as_downcasting();
+
+    let mut options = CollatorOptions::new();
+    options.set_strength(Some(Strength::Tertiary));
+
+    {
+        let collator: Collator = Collator::try_new(locale.clone(), &provider_no_fallback, options).unwrap();
+        check_expectations(&collator, &left, &right, &expectations);
+    }
+
+    options.set_strength(Some(Strength::Primary));
+
+    {
+        let collator: Collator = Collator::try_new(locale, &provider_no_fallback, options).unwrap();
+        check_expectations(&collator, &left, &right, &expectations);
+    }
+}
+
+#[test]
+fn test_nb_nn_no() {
+    let any_dyn_provider = get_provider();
+    let any_provider = any_dyn_provider.as_any_provider();
+    let provider_with_fallback = LocaleFallbackProvider::try_new(any_provider.as_downcasting()).unwrap();
+    let provider_no_fallback = any_provider.as_downcasting();
+
+    let result: Result<DataResponse<CollationDataV1Marker>, DataError> = provider_no_fallback.load_resource(&DataRequest {
+        options: locale!("nb").into(),
+        metadata: Default::default()
+    });
+    assert!(matches!(result, Err(_)));
+
+    let data_response: DataResponse<CollationDataV1Marker> = provider_with_fallback.load_resource(&DataRequest {
+        options: locale!("nb").into(),
+        metadata: Default::default()
+    }).unwrap();
+    assert_eq!(data_response.metadata.data_locale, Some(locale!("no").into()));
```

Comment:
@hsivonen My test here isn't finding "no" data at all. What key should I use, or could you suggest how to test from a Collator API level that "nb" and "no" have the same behavior?

Reply:
We don't currently have collation data for Norwegian in the repo, so this code either needs to work with the full zip file or additional data from the zip needs to be added to https://github.com/unicode-org/icu4x/tree/main/provider/testdata/data/icuexport/collation/implicithan

As for what behavior to test, I suggest testing that "ü", "y", "å", "ø" sort into "y", "ü", "ø", "å".

Reply:
Looking at what the code does, it seems the answer is to add the `no_*` collation files to the repo.

Reply:
And the option with the best developer ergonomics would, of course, be to put the full set of collation data files in the repo and not worry about the slight repo size increase, since that wouldn't affect the `testdata.postcard` size, but would make tests like these Just Work.

Reply:
Yes; I will add the necessary files to the repo without adding them to testdata.postcard. I will do that in a follow-up PR.